### DISCUSSION TITLE
Iframe WebXR issue

### DISCUSSION THREAD

**Discussion by danilokrulj** [google/model-viewer#4642] [Q&A] [answered]
I'm having iframe issues where the content we made is delivered through an iframe and the page opens SceneViewer all the time instead of WebXR, the permissions are supposedly set on the clients side but im an not 100% sure they are set as necessary on our side. Can anyone help me with this, how can i be sure that the permissions are causing the issue.

**Comment by elalish** [comment]
Did you follow these instructions? https://medium.com/samsung-internet-dev/using-webxr-in-html-iframes-629c248ff96d

Also make sure the phone has accepted WebXR permissions for that page - you may need to clear the permissions if anyone has ever clicked "deny".

**Reply by danilokrulj** [reply]
Yes, everything is on https and i have the allow=”xr-spatial-tracking” in the iframe too but it doesn't seem to make a difference. The client set up the main pages header to also contain the xr-spatial-tracking permission policy.

**Reply by danilokrulj** [reply]
Maybe its worth saying that when the iframe and the content are on the same domain, WebXR works. When the content is on one domain and the iframe is on another, SceneViewer is opened instead of WebXR. I have tested this with multiple domains and the results were the same.

**Reply by elalish** [reply]
Interesting - can you link us to a page where we can repro your cross-origin problem? It's possible this is a Chrome bug.

**Reply by elalish** [reply, marked as answer]
Also, can you try remixing the Glitch in that article? It demonstrates cross-origin WebXR, so maybe that code will help you find the problem.

**Reply by danilokrulj** [reply]
Looks like I was just missing allowfullscreen="" in my iframe... Thank you @elalish for sugesting remixing the Glitch, I didn't think it would be something as simple as this.
